src/overlays/actors/ovl_En_Sth/z_en_sth.c

```diff
@@ -40,45 +55,357 @@ static ColliderCylinderInit D_80B0B404 = {
+};
+
+static AnimationHeader* sAnimations[6] = {
+    &gParentDanceAnim, &gChildDanceAnim, &gChildDanceAnim, &gChildDanceAnim, &gChildDanceAnim, &gChildDanceAnim,
+};
+
+static EnSthActionFunc sRewardObtainedWaitActions[6] = {
+    EnSth_ParentRewardObtainedWait, EnSth_ChildRewardObtainedWait, EnSth_ChildRewardObtainedWait,
+    EnSth_ChildRewardObtainedWait,  EnSth_ChildRewardObtainedWait, EnSth_ChildRewardObtainedWait,
+};
+
+static u16 sEventFlags[6] = {
+    0x0000, 0x0400, 0x0800, 0x1000, 0x2000, 0x4000,
+};
+
+static s16 sGetItemIds[6] = {
+    GI_RUPEE_GOLD, GI_WALLET_ADULT, GI_STONE_OF_AGONY, GI_WALLET_GIANT, GI_BOMBCHUS_10, GI_HEART_PIECE,
+};
+
+static Vec3f D_80B0B49C = { 700.0f, 400.0f, 0.0f };
+
+static Color_RGB8 sTunicColors[6] = {
+    { 190, 110, 0 }, { 0, 180, 110 }, { 0, 255, 80 }, { 255, 160, 60 }, { 190, 230, 250 }, { 240, 230, 120 },
+};
+
+void EnSth_SetupAction(EnSth* this, EnSthActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnSth_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnSth* this = THIS;
+
+    s16 objectId;
+    s32 params = this->actor.params;
+    s32 objectBankIdx;
+    s32 params2;
+
+    // Translation: Gold Skulltula Shop
+    osSyncPrintf(VT_FGCOL(BLUE) "金スタル屋 no = %d\n" VT_RST, params);
+    params2 = this->actor.params;
+    if (params2 == 0) {
+        if (gSaveContext.inventory.gsTokens < 100) {
+            Actor_Kill(&this->actor);
+            // Translation: Gold Skulltula Shop I still can't be a human
+            osSyncPrintf("金スタル屋 まだ 人間に戻れない \n");
+            return;
+        }
+    } else if (gSaveContext.inventory.gsTokens < (params2 * 10)) {
+        Actor_Kill(&this->actor);
+        // Translation: Gold Skulltula Shop I still can't be a human
+        osSyncPrintf(VT_FGCOL(BLUE) "金スタル屋 まだ 人間に戻れない \n" VT_RST);
+        return;
+    }
+
+    objectId = sObjectIds[params];
+    if (objectId != 1) {
+        objectBankIdx = Object_GetIndex(&globalCtx->objectCtx, objectId);
+    } else {
+        objectBankIdx = 0;
+    }
+
+    osSyncPrintf("bank_ID = %d\n", objectBankIdx);
+    if (objectBankIdx < 0) {
+        __assert("0", "../z_en_sth.c", 1564);
+    }
+    this->objectBankIdx = objectBankIdx;
+    this->drawFunc = EnSth_Draw;
+    Actor_SetScale(&this->actor, 0.01f);
+    EnSth_SetupAction(this, EnSth_WaitForObjectLoaded);
+    this->actor.draw = NULL;
+    this->unk_2B2 = 0;
+    this->actor.targetMode = 6;
+}
+
+void EnSth_SetupCollider(EnSth* this, GlobalContext* globalCtx) {
+    s32 pad;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 36.0f);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->actor.update = EnSth_Update2;
+    this->actor.draw = this->drawFunc;
+}
+
+void EnSth_SetupAnimation(EnSth* this, GlobalContext* globalCtx) {
```

Comment:
These seem to do more than just setup the collider and animations. Perhaps the names could be improved.